### PR TITLE
Add ws image buffer blit

### PR DESCRIPTION
**Pull request by TheNeikos**
This pr implements `ws_buffer_blit()`

### CODE REVIEW

File: src/compositor/buffer.h [hunk 1]
```diff
@@ -177,6 +177,18 @@ ws_buffer_end_access(
 )
 __ws_nonnull__(1)
 ;
+/**
+ * Blit two buffers together (This copies one into the other)
+ *
+ * @warning to not pass NULL to this function! It will crash!
```
Comment (matthiasbeyer) [inline src/compositor/buffer.h]: "to" not? :-)

File: src/compositor/buffer.c [hunk 2]
```diff
@@ -185,3 +188,26 @@ ws_buffer_end_access(
...
+    }
+
+    int max_x = MAX(ws_buffer_width(dest), ws_buffer_width(src));
+    int max_y = MAX(ws_buffer_height(dest), ws_buffer_height(src));
+
+    int stride_dst = ws_buffer_stride(dest);
+    int stride_src = ws_buffer_stride(src);
+    for (int y = 0; y < max_y; ++y) {
+        memcpy((char*)buf_dst + (y * stride_dst),
+                (char*)buf_src + (y * stride_src),
```
Comment (matthiasbeyer) [inline src/compositor/buffer.c]: This is supposed to be `(char*) (a + b)` , or do you mean `((char*) a) + b` ?

Comment (TheNeikos) [inline src/compositor/buffer.c]: No, you first cast it to a char, so that you can actually do pointer arithmetic on it.

Comment (matthiasbeyer) [inline src/compositor/buffer.c]: So, the second one.

Please add some parentheses, so one does not have to think about operator preference and this stuff!

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
ACK. @neithernut wanna merge?

**Pull request merged**